k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -0,0 +1,276 @@
+#include "converters/EDM4hep2Lcio.h"
+
+
+DECLARE_COMPONENT(EDM4hep2LcioTool);
+
+// Add EDM4hep to LCIO converted tracks to vector
+void EDM4hep2LcioTool::addLCIOConvertedTracks(
+  std::vector<std::pair<lcio::TrackImpl*, edm4hep::Track>>& lcio_tracks_vec,
+  const std::string& name,
+  const std::string& lcio_collection_name,
+  lcio::LCEventImpl* lcio_event)
+{
+  DataHandle<edm4hep::TrackCollection> tracks_handle {
+    name, Gaudi::DataHandle::Reader, this};
+  const auto tracks_coll = tracks_handle.get();
+
+  auto* tracks = new lcio::LCCollectionVec(lcio::LCIO::TRACK);
+
+  // Loop over EDM4hep tracks converting them to lcio tracks
+  for (int i = 0; i < tracks_coll->size(); ++i) {
+
+    const edm4hep::Track edm_tr = (*tracks_coll)[i];
+
+    auto* lcio_tr = new lcio::TrackImpl();
+
+    lcio_tr->setTypeBit( edm_tr.getType() );
+    lcio_tr->setChi2( edm_tr.getChi2() );
+    lcio_tr->setNdf( edm_tr.getNdf() );
+    lcio_tr->setdEdx( edm_tr.getDEdx() );
+    lcio_tr->setdEdxError( edm_tr.getDEdxError() );
+    lcio_tr->setRadiusOfInnermostHit( edm_tr.getRadiusOfInnermostHit() );
```

Comment:
```suggestion
    lcio_tr->setRadiusOfInnermostHit( edm_tr.getRadiusOfInnermostHit() );
    // FIXME, correctly assign hitnumbers
    lcio_tr->subdetectorHitNumbers().resize(50);
    for(int i =0;i<50;++i) {
      lcio_tr->subdetectorHitNumbers()[i] = 0 ; 
    }
```

Reply:
Maybe the conversion from edm4hep was forgotten?
https://github.com/key4hep/EDM4hep/blob/1bd0b5db00ec8659c53041e07784c34f8cc01add/edm4hep.yaml#L325

Reply:
fixed